crates/shared/src/price_estimation/trade_verifier/mod.rs

```diff
@@ -177,28 +176,34 @@ impl TradeVerifier {
             )
             .tx;
 
-        let swap_simulation = Solver::Solver::swapCall {
-            settlementContract: self.settlement.address().into_alloy(),
-            tokens: tokens.iter().cloned().map(IntoAlloy::into_alloy).collect(),
-            receiver: verification.receiver.into_alloy(),
-            settlementCall: alloy::primitives::Bytes::from(settlement.data.unwrap().0),
-        };
-
         let block = *self.block_stream.borrow();
 
-        let call = CallRequest {
+        let solver = Solver::Instance::new(solver_address.into_alloy(), self.web3.alloy.clone());
+        let swap_simulation = solver.swap(
+            self.settlement.address().into_alloy(),
+            tokens.iter().cloned().map(IntoAlloy::into_alloy).collect(),
+            verification.receiver.into_alloy(),
+            alloy::primitives::Bytes::from(settlement.data.unwrap().0),
+        )
             // Initiate tx as solver so gas doesn't get deducted from user's ETH.
-            from: Some(solver_address),
-            to: Some(solver_address),
-            data: Some(swap_simulation.abi_encode().into()),
-            gas: Some(Self::DEFAULT_GAS.into()),
-            gas_price: Some(block.gas_price),
-            ..Default::default()
-        };
+        .from(solver_address.into_alloy())
+        .to(solver_address.into_alloy())
+        .gas(Self::DEFAULT_GAS)
+        .gas_price(u128::try_from(block.gas_price).map_err(|err| anyhow!(err)).context("converting gas price to u128")?);
+
+        if let Some(tenderly) = &self.simulator
+            && let Err(err) = tenderly.log_simulation_command(
+                swap_simulation.clone().into_transaction_request(),
+                overrides.clone(),
+                Some(block.number),
```

Comment:
nit: to get rid of the lonely line 179 above.
```suggestion
                Some(*self.block_stream.borrow()),
```